gpkit/constraints/gp.py

```diff
@@ -73,16 +74,13 @@ def __init__(self, cost, constraints, substitutions=None, verbosity=1):
         # sideways NomialData init to create self.exps, self.cs, etc
         self.posynomials = [cost.sub(self.substitutions)]
         self.posynomials.extend(self.as_posyslt1(self.substitutions))
-        self.gen(verbosity)
-
-    def gen(self, verbosity=1):
-        "Generates nomial and solve data (A, p_idxs) from self.posynomials"
-        NomialData.init_from_nomials(self, self.posynomials)
-        if self.any_nonpositive_cs:
+        self.hmaps = [p.hmap for p in self.posynomials]
+        self.gen()
+        if any(c <= 0 for c in self._cs):
```

Comment:
Not going to add these for now because `gen()` is only designed for fast SPs, but creating an issue linking to this for when gen is refactored